bll/Services/AccountService.cs

```diff
@@ -0,0 +1,101 @@
+using dal.Context;
+using dal.Models;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.Extensions.Configuration;
+using Microsoft.IdentityModel.Tokens;
+using System;
+using System.Collections.Generic;
+using System.IdentityModel.Tokens.Jwt;
+using System.Linq;
+using System.Security.Claims;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace bll.Services
+{
+    public class AccountService : IAccountService
+    {
+        private readonly UserManager<User> userManager;
+        private readonly RoleManager<ApplicationRole> roleManager;
+        private readonly IConfiguration configuration;
+        private readonly IMapper mapper;
+
+        public AccountService(UserManager<User> userManager, RoleManager<ApplicationRole> roleManager, IConfiguration configuration, IMapper mapper)
+        {
+            this.configuration = configuration;
+            this.userManager = userManager;
+            this.roleManager = roleManager;
+            this.mapper = mapper;
+        }
+
+        public async Task<string> Login(LoginModel userLogin)
+        {
+            var user = await userManager.FindByNameAsync(userLogin.UserName);
+            var checkPassword = await userManager.CheckPasswordAsync(user, userLogin.Password);
+
+            if (user == null) throw new NullReferenceException();
+            if (!checkPassword) throw new Exception("Wrong password");
+            if (DateTime.UtcNow < user.LockoutEnd) throw new Exception("User is blocked");
+
+            var userRoles = await userManager.GetRolesAsync(user);
+            var authClaims = new List<Claim>
+            {
+                new Claim(ClaimTypes.NameIdentifier, user.UserName)
+            };
+
+            authClaims.AddRange(userRoles.Select(role => new Claim(ClaimTypes.Role, role)));
+            var token = GetJwtSecurityToken(authClaims);
+            return token;
+        }
+
+        public async Task<UserDto> Register(RegisterModel userRegister)
+        {
+            var userExists = await userManager.FindByNameAsync(userRegister.UserName);
+            if (userExists != null) {
+                throw new Exception("User name is already exists");
+            }
+
+            if (userRegister.Role != UserRoles.Admin || userRegister.Role != UserRoles.User) {
+                throw new InvalidOperationException("Role doesn't exist");
+            }
+
+            var user = new User()
+            {
+                FirstName = userRegister.FirstName,
+                LastName = userRegister.LastName,
+                DateOfBirth = userRegister.DateOfBirth,
+                UserName = userRegister.UserName,
+                Password = userRegister.Password,
+            };
+            var result = await userManager.CreateAsync(user, userRegister.Password);
+            if (!result.Succeeded) {
+                throw new Exception("User isn't created");
+            }
+
+            if (!await roleManager.RoleExistsAsync(UserRoles.Admin) ||
```

Comment:
What if Admin role exists but User doesn't? Or Admin not exists, but user does? 
This is a bad idea anyway. If you want to make roles static, than you shouldn't store them in db. If you want them dynamic, than you should not hardcode them. If having user roles in db is required by the framework such as AspNetIdentity, you can check the database roles on startup. But... If the framework requires something you don't want, why would you use it?